test/formulas/Utils.js

```diff
@@ -0,0 +1,177 @@
+const {FormulaParser} = require('../../grammar/hooks');
+const FormulaError = require('../../formulas/error');
+const {DepParser} = require('../../grammar/dependency/hooks');
+const {MAX_ROW, MAX_COLUMN} = require('../../index');
+const Utils = {
+    getParser(test){
+        const data = test["sheet"]
+        if(data){
+            data.forEach((row, rowIndex) => row.forEach((value, colIndex) => {
+                if(value === "undefined")
+                    data[rowIndex][colIndex] = undefined
+                if(FormulaError.errorMap.has(value))
+                    data[rowIndex][colIndex] = FormulaError.errorMap.get(value)
+            }));
+        }
+        const map = {
+            "formula" : new FormulaParser({
+                onCell: ref => {
+                    return test["sheet"][ref.row - 1][ref.col - 1]
+                },
+                onRange: ref => {
+                    const arr = [];
+                    for (let row = ref.from.row - 1; row < ref.to.row; row++) {
+                        const innerArr = [];
+                        for (let col = ref.from.col - 1; col < ref.to.col; col++) {
+                            innerArr.push(test["sheet"][row][col])
+                        }
+                        arr.push(innerArr);
+                    }
+                    return arr;
+                },
+                onVariable: name => {
+                    return test["variables"][name]
+                },
+            }),
+            "depParser": new DepParser({
+                onVariable: variable => {
```

Comment:
Each test should have a variable map from row,col pairs to values.
 onVariable calls things from that map. For now just use a dumb list, since it's O(n) and n is very small.